service/matching/matching_engine.go

```diff
@@ -1249,16 +1250,80 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 			},
 		}, nil
 	}
+
 	// Otherwise, do legacy DescribeTaskQueue
 	partition, err := tqid.PartitionFromProto(req.TaskQueue, request.GetNamespaceId(), req.TaskQueueType)
 	if err != nil {
 		return nil, err
 	}
-	pm, _, err := e.getTaskQueuePartitionManager(ctx, partition, true, loadCauseDescribe)
+	rootPM, _, err := e.getTaskQueuePartitionManager(ctx, partition, true, loadCauseDescribe)
+	if err != nil {
+		return nil, err
+	}
+	//nolint:staticcheck // SA1019 deprecated
+	descrResp, err := rootPM.LegacyDescribeTaskQueue(req.GetIncludeTaskQueueStatus())
 	if err != nil {
 		return nil, err
 	}
-	return pm.LegacyDescribeTaskQueue(req.GetIncludeTaskQueueStatus())
+
+	if req.ReportStats {
+		cacheKey := ".taskQueueStats" // starts with `.` to prevent clashing with build ID cache keys
+		if ts := rootPM.GetCache(cacheKey); ts != nil {
+			//revive:disable-next-line:unchecked-type-assertion
+			descrResp.DescResponse.TaskQueueStats = ts.(*taskqueuepb.TaskQueueStats)
+		} else {
+			taskQueueStats := &taskqueuepb.TaskQueueStats{}
+
+			// find all buildIds
+			var buildIds []string
+			userData, _, err := rootPM.GetUserDataManager().GetUserData()
+			if err != nil {
+				return nil, err
+			}
+			if userData != nil {
```

Comment:
you don't need this line either

Reply:
Hm, I could also swear I added that after a NPE. But I can't recall anymore.

Reply:
removed it now 👍